components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt

```diff
@@ -369,18 +370,48 @@ class TabsUseCases(
     class SelectOrAddUseCase(
         private val store: BrowserStore
     ) {
+        /**
+         * Selects an already existing tab with the matching [HistoryMetadataKey] or otherwise
+         * creates a new tab with the given [url].
+         *
+         * @param url The URL to be selected or loaded in the new tab.
+         * @param historyMetadata The [HistoryMetadataKey] to match for existing tabs.
+         * @return The ID of the selected or created tab.
+         */
+        operator fun invoke(
+            url: String,
+            historyMetadata: HistoryMetadataKey
+        ): String {
+            val tab = store.state.tabs.find { it.historyMetadata == historyMetadata }
+
+            return if (tab != null) {
+                store.dispatch(TabListAction.SelectTabAction(tab.id))
+                tab.id
+            } else {
+                this.invoke(url)
+            }
+        }
+
         /**
          * Selects an already existing tab displaying [url] or otherwise creates a new tab.
+         *
+         * @param url The URL to be loaded in the new tab.
+         * @param private Whether or not this session should use private mode.
```

Comment:
This is actually a bug for the found and selected case but we have it on file: https://github.com/mozilla-mobile/android-components/issues/10560 to address outside of this.